package-res/ccc/core/base/chart/chart.panels.js

```diff
@@ -274,17 +310,34 @@ pvc.BaseChart
      * @virtual
      */
     _createContent: function(parentPanel, contentOptions) {
+        var index = 0;
+
         this.plotList.forEach(function(plot) {
-            this._createPlotPanel(plot, parentPanel, contentOptions);
+            this._createPlotPanel(plot, parentPanel, contentOptions, index);
+            index++; //CDF603 added index information to plots: position in plotList, assuming it does not change
         }, this);
     },
 
-    _createPlotPanel: function(plot, parentPanel, contentOptions) {
-        var PlotPanelClass = pvc.PlotPanel.getClass(plot.type);
+    _createPlotPanel: function(plot, parentPanel, contentOptions, index) {
+        var PlotPanelClass = pvc.PlotPanel.getClass(plot.type);        
+
         if(!PlotPanelClass)
             throw def.error.invalidOperation("There is no registered panel class for plot type '{0}'.", [plot.type]);
 
-        var panel = new PlotPanelClass(this, parentPanel, plot, Object.create(contentOptions)),
+        var opts     = Object.create(contentOptions);
+
+        // CDF603
+        /* Add preserved plot layout info to options if the layout should be preserved
+           This uses the index in list, assuming the order is the same
+        */
+        if(this._preserveLayout){
+            var infoPrev = this.preservedPlotsLayoutInfoList[index];
```

Comment:
Remove TODO. Assume by index and that structure doesn't change, so the "infoPrev" tests are also not needed.